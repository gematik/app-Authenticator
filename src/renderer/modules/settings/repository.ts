/*
 * Copyright 2023 gematik GmbH
 *
 * The Authenticator App is licensed under the European Union Public Licence (EUPL); every use of the Authenticator App
 * Sourcecode must be in compliance with the EUPL.
 *
 * You will find more details about the EUPL here: https://joinup.ec.europa.eu/collection/eupl
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the EUPL is distributed on an "AS
 * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the EUPL for the specific
 * language governing permissions and limitations under the License.ee the Licence for the specific language governing
 * permissions and limitations under the Licence.
 */

import dot from 'dot-object';
import Swal from 'sweetalert2';

// #!if MOCK_MODE === 'ENABLED'
import { DEVELOPER_OPTIONS, MOCK_CONNECTOR_CONFIG } from '@/renderer/modules/connector/connector-mock/mock-config';
// #!endif
import {
  APP_NAME,
  CONFIG_FILE_NAME,
  IPC_GET_PATH,
  IPC_READ_CREDENTIALS,
  IPC_SAVE_CREDENTIALS,
  PROCESS_ENVS,
  PRODUCT_NAME,
} from '@/constants';
import { logger } from '@/renderer/service/logger';
import {
  CHECK_UPDATES_AUTOMATICALLY_CONFIG,
  ENTRY_OPTIONS_CONFIG_GROUP,
  PROXY_SETTINGS_CONFIG,
  SENSITIVE_CONFIG_KEYS_LIST,
  TLS_AUTH_TYPE_CONFIG,
} from '@/config';
import { TlsAuthType } from '@/@types/common-types';
import { ERROR_CODES } from '@/error-codes';
import { UserfacingError } from '@/renderer/errors/errors';
import i18n from '@/renderer/i18n';

export interface TRepositoryData {
  [key: string]: number | string | boolean;
}

/**
 * Holds data in the cache to prevent reading the config file frequently
 */
let storedData: TRepositoryData = {};

export interface ISettingsRepository {
  save(data: TRepositoryData): void;

  load(forceReload?: boolean): TRepositoryData;

  clear(): void;

  exist(): boolean;

  /**
   * This only updates the cached config and does not change the config.json
   * This is relevant for making function tests on the fly
   */
  setWithoutSave(data: TRepositoryData): void;
}

export const INITIAL_STATE = {
  [CHECK_UPDATES_AUTOMATICALLY_CONFIG]: true,
  [PROXY_SETTINGS_CONFIG.AUTH_TYPE]: false,
  [PROXY_SETTINGS_CONFIG.USE_OS_SETTINGS]: true,
  [ENTRY_OPTIONS_CONFIG_GROUP.TLS_REJECT_UNAUTHORIZED]: false,
  [TLS_AUTH_TYPE_CONFIG]: TlsAuthType.ServerCertAuth,

  // #!if MOCK_MODE === 'ENABLED'
  [MOCK_CONNECTOR_CONFIG]: false,
  [DEVELOPER_OPTIONS.IDP_CERTIFICATE_CHECK]: true,
  // #!endif
};

export class FileStorageRepository implements ISettingsRepository {
  static get isNewInstallation(): boolean {
    return this._isNewInstallation;
  }
  static set isNewInstallation(value: boolean) {
    this._isNewInstallation = value;
  }

  static get usesCredentialManager(): boolean {
    return this._usesCredentialManager;
  }

  static set usesCredentialManager(value: boolean) {
    this._usesCredentialManager = value;
  }

  static get isJsonFileInvalid(): boolean {
    return this._isJsonFileInvalid;
  }

  static set isJsonFileInvalid(value: boolean) {
    this._isJsonFileInvalid = value;
  }

  private static encoding: BufferEncoding = 'utf-8';
  private static _path: string | null = null;
  private static _usesCredentialManager = false;
  private static _isNewInstallation = false;
  private static _isJsonFileInvalid = false;

  public static getConfigDir(): { path: string; localEnv: boolean } {
    const clientName = PROCESS_ENVS.CLIENTNAME;
    const computerName = PROCESS_ENVS.COMPUTERNAME;
    const authConfigPath = PROCESS_ENVS.AUTHCONFIGPATH;
    const viewClientMachineName = PROCESS_ENVS.VIEWCLIENT_MACHINE_NAME || PROCESS_ENVS.ViewClient_Machine_Name;

    logger.info('configPath:');
    logger.info('  -  AUTHCONFIGPATH:' + authConfigPath);
    logger.info('  -  CLIENTNAME:' + clientName);
    logger.info('  -  COMPUTERNAME:' + computerName);
    logger.info('  -  VIEWCLIENT_MACHINE_NAME:' + viewClientMachineName);

    // Abhängig von folgenden Umgebungsvariablen wird die Lokalität der zu verwendenden config.json in folgender Reihenfolge bestimmt:
    // 1. Wenn die Umgebungsvariable AUTHCONFIGPATH gesetzt ist und die Umgebungsvariable ViewClient_Machine_Name (VMWARE)
    //    → config.json im Folder: AUTHCONFIGPATH + //ViewClient_Machine_Name//config.json
    // 2. Wenn die Umgebungsvariable AUTHCONFIGPATH gesetzt ist und die Umgebungsvariable CLIENTNAME (CITRIX)
    //    → config.json im Folder: AUTHCONFIGPATH + //CLIENTNAME//config.json
    // 3. Wenn die Umgebungsvariable AUTHCONFIGPATH gesetzt ist und die Umgebungsvariable COMPUTERNAME
    //    → config.json im Folder: AUTHCONFIGPATH + //COMPUTERNAME//config.json
    // 4. Für alle anderen Fälle (auch im Fehlerfall das der AUTHCONFIGPATH nicht existiert!):
    //    → config.json im Installationsverzeichnis der exe (wie bisher)

    //wenn keine config.json in dem viewClientMachineName, clientName oder computerName Pfad existiert, dann wird die lokale config benutzt
    let configPath = '';
    if (authConfigPath) {
      if (viewClientMachineName) {
        logger.info(' - config über ViewClient_Machine_Name');
        configPath = window.api.pathJoin(authConfigPath.toString(), viewClientMachineName.toString());
      } else if (clientName) {
        logger.info(' - config über CLIENTNAME');
        configPath = window.api.pathJoin(authConfigPath.toString(), clientName.toString());
      } else if (computerName) {
        logger.info(' - config über COMPUTERNAME');
        configPath = window.api.pathJoin(authConfigPath.toString(), computerName.toString());
      }
      return { path: configPath, localEnv: false };
    }

    logger.info(' - config über Anwendungsverzeichnis');
    return { path: window.api.sendSync(IPC_GET_PATH, 'userData') as string, localEnv: true };
  }

  public static getPath(): string {
    if (FileStorageRepository._path) {
      return FileStorageRepository._path;
    }

    const config = FileStorageRepository.getConfigDir();
    let path = config.path;
    if (config.localEnv) {
      path = path.replace('Roaming', 'Local');

      //C:\Users\USERNAME\AppData\Local\authenticator -> C:\Users\USERNAME\AppData\Local\gematik Authenticator
      path = path.replace(window.api.pathSep() + APP_NAME, window.api.pathSep() + PRODUCT_NAME);

      if (!window.api.existsSync(path)) {
        try {
          window.api.mkdirSync(path);
        } catch (err) {
          logger.error('Config dir could not be created', err.message);
        }
      }
    }

    path += window.api.pathJoin('/', CONFIG_FILE_NAME);

    FileStorageRepository._path = path;

    logger.info(' - config.json path: ' + path);
    return path;
  }

  save(data: TRepositoryData): void {
    // update state
    storedData = dot.dot(data);

    // send data to preload & main
    window.api.setAppConfigInPreload(storedData);

    const { shouldSaveToCM, showWarningOnFail, saveAllToConfigFileOnFail } = FileStorageRepository.decideToSaveToCM();
    const nonSensitiveData = FileStorageRepository.filterSensitiveData(data);
    let dataToSaveConfigFile: TRepositoryData = saveAllToConfigFileOnFail ? data : nonSensitiveData;

    // save to credential manager
    if (shouldSaveToCM) {
      const savedToCMSuccessfully = FileStorageRepository.saveToCm(data);

      if (!savedToCMSuccessfully && showWarningOnFail) {
        throw new UserfacingError('Could not save to credential manager', '', ERROR_CODES.AUTHCL_0010);
      }

      if (savedToCMSuccessfully) {
        dataToSaveConfigFile = nonSensitiveData;
      }
    } else {
      dataToSaveConfigFile = data;
    }

    const unFlatted = dot.object({ ...dataToSaveConfigFile });
    window.api.writeFileSync(FileStorageRepository.getPath(), JSON.stringify(unFlatted, null, 2));

    logger.info('Saved config changes under: ', FileStorageRepository.getPath());

    // #!if MOCK_MODE === 'ENABLED'
    FileStorageRepository.printConfig();
    // #!endif
  }

  static saveToCm(data?: TRepositoryData): boolean {
    return window.api.sendSync(IPC_SAVE_CREDENTIALS, data) as boolean;
  }

  static readFromCm(): Partial<TRepositoryData> {
    return (window.api.sendSync(IPC_READ_CREDENTIALS) as Partial<TRepositoryData>) || {};
  }

  static decideToSaveToCM(): {
    shouldSaveToCM: boolean;
    showWarningOnFail: boolean;
    saveAllToConfigFileOnFail: boolean;
  } {
    let shouldSaveToCM;
    let showWarningOnFail;
    let saveAllToConfigFileOnFail;

    const isCentralConfiguration = PROCESS_ENVS.AUTHCONFIGPATH;

    // If it's a new installation
    if (FileStorageRepository._isNewInstallation) {
      shouldSaveToCM = true;
      showWarningOnFail = true;
      saveAllToConfigFileOnFail = false;
    } else {
      if (isCentralConfiguration) {
        if (FileStorageRepository._usesCredentialManager) {
          shouldSaveToCM = true;
          saveAllToConfigFileOnFail = false;
          showWarningOnFail = true;
        } else {
          shouldSaveToCM = false;
          saveAllToConfigFileOnFail = true;
          showWarningOnFail = true;
        }
      } else {
        saveAllToConfigFileOnFail = false;
        showWarningOnFail = true;
        shouldSaveToCM = true;
      }
    }

    return { shouldSaveToCM, showWarningOnFail, saveAllToConfigFileOnFail };
  }
  /**
   * This function filters sensitive data from the config data
   * @param data
   */
  static filterSensitiveData(data: TRepositoryData): TRepositoryData {
    const nonSensitiveData: TRepositoryData = {};

    for (const key in data) {
      if (!SENSITIVE_CONFIG_KEYS_LIST.includes(key)) {
        nonSensitiveData[key] = data[key];
      }
    }

    return nonSensitiveData;
  }

  /**
   * todo be sure that config file exists, and has a right format. If not, throw error
   */
  load(forceReload = false): TRepositoryData {
    if (!forceReload && Object.keys(storedData).length) {
      return storedData;
    }

    // forceReload happens when env vars get changed;
    // that means we need to recalculate the path
    if (forceReload) {
      FileStorageRepository._path = null;
    }

    const sensitiveDataFromCredentialsManager = FileStorageRepository.readFromCm();

    // set usesCredentialManager if sensitiveDataFromCredentialsManager is not empty
    FileStorageRepository._usesCredentialManager = !!Object.keys(sensitiveDataFromCredentialsManager).length;

    if (!window.api.existsSync(FileStorageRepository.getPath())) {
      FileStorageRepository._isNewInstallation = true;
      return { ...INITIAL_STATE };
    }

    const buffer = window.api.readFileSync(FileStorageRepository.getPath(), FileStorageRepository.encoding);
    const string = buffer.toString();

    let json: TRepositoryData;

    try {
      // convert string to json and convert string booleans to booleans
      json = JSON.parse(string, (_, value) => {
        if (value === 'true') return true;
        if (value === 'false') return false;
        return value;
      });
      FileStorageRepository._isJsonFileInvalid = false;
    } catch (e) {
      logger.error('Error while parsing config file', e);
      FileStorageRepository._isJsonFileInvalid = true;

      Swal.fire({
        icon: 'error',
        title: i18n.global.t(`errors.${ERROR_CODES.AUTHCL_0012}.title`),
        text: i18n.global.t(`errors.${ERROR_CODES.AUTHCL_0012}.text`, { path: FileStorageRepository.getPath() }),
      });

      return {};
    }

    const dottedData = {
      ...dot.dot(json),
      // get sensitive data from credential manager
      ...sensitiveDataFromCredentialsManager,
    };

    const data = {
      ...INITIAL_STATE,
      ...dottedData,
    };
    storedData = data;

    // send data to preload & main
    window.api.setAppConfigInPreload(storedData);
    // #!if MOCK_MODE === 'ENABLED'
    FileStorageRepository.printConfig();
    // #!endif
    return data;
  }

  clear(): void {
    storedData = {};

    if (window.api.existsSync(FileStorageRepository.getPath())) {
      window.api.unlinkSync(FileStorageRepository.getPath());
    }
  }

  exist(): boolean {
    return window.api.existsSync(FileStorageRepository.getPath());
  }

  /**
   * This only updates the cached config and does not change the config.json
   * @param data
   */
  setWithoutSave(data: TRepositoryData): void {
    storedData = data;
  }

  // #!if MOCK_MODE === 'ENABLED'
  private static printConfig() {
    logger.info('print settings:');
    for (const item in storedData) {
      if (!item.toLowerCase().includes('password')) {
        logger.info('  - ' + item + ':' + storedData[item]);
      }
    }
  }

  // #!endif
}
