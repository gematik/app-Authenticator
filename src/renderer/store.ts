/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 *
 * ******
 *
 * For additional notes and disclaimer from gematik and in case of changes by gematik find details in the "Readme" file.
 */

import { createStore } from 'vuex';
import { connectorStore } from '@/renderer/modules/connector/store/connector-store';
import { TConnectorStore } from '@/renderer/modules/connector/type-definitions';
import { TIdpServiceStore } from '@/renderer/modules/gem-idp/store/idp-service-store.d';
import { idpServiceStore } from '@/renderer/modules/gem-idp/store/idp-service-store';

export type TRootStore = {
  connectorStore: TConnectorStore;
  idpServiceStore: TIdpServiceStore;
  settingsIsSet: boolean;
  showLoadingSpinner: boolean;
};

export default createStore({
  state: (): TRootStore => {
    return {
      showLoadingSpinner: false,
    } as TRootStore;
  },
  mutations: {
    setShowLoadingSpinner(state, newState: boolean) {
      state.showLoadingSpinner = newState;
    },
  },
  actions: {},
  modules: {
    connectorStore,
    idpServiceStore,
  },
});
