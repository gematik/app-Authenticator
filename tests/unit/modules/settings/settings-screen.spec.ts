/*
 * Copyright 2025, gematik GmbH
 *
 * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by the
 * European Commission â€“ subsequent versions of the EUPL (the "Licence").
 * You may not use this work except in compliance with the Licence.
 *
 * You find a copy of the Licence in the "Licence" file or at
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expressed or implied.
 * In case of changes by gematik find details in the "Readme" file.
 *
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 */

import { mount } from '@vue/test-utils';
import { SettingsScreen } from '@/renderer/modules/settings';
import i18n from '@/renderer/i18n';
import store from '@/renderer/store';
import { FileStorageRepository } from '@/renderer/modules/settings/repository';

const fileStorageRepository = new FileStorageRepository();
jest.mock('@/renderer/modules/settings/useSettings.ts', () => ({
  useSettings: () => {
    return fileStorageRepository;
  },
}));

describe('settings screen', () => {
  it('render ', async function () {
    const wrapper = mount(SettingsScreen, {
      global: {
        plugins: [store, i18n],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
