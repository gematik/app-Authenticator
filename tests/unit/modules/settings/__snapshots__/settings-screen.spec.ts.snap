// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings screen render  1`] = `
<div
  data-v-app=""
>
  
  <div
    class="antialiased w-full justify-center"
  >
    <form
      class="w-full"
    >
      
      <div
        class="w-full"
      >
        <div>
          <div
            class="flex items-center w-full bg-white border-border border-[1px] rounded-t-[8px] h-[72px] py-[26px] px-[48px]"
          >
            <!--v-if-->
            <h3
              class=""
              id="lblTitle-0"
            >
              Mock Konnektor
            </h3>
          </div>
          
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.mockConnector"
            >
              <label
                class="text-sm"
                id="form-connector.mockConnector-label"
              >
                Status
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Durch diese Option wird die Hardware, wie der Konnektor, das Kartenterminal und die Smartcards simuliert und ist nur für Entwickler und Tester vorgesehen
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-connector.mockConnector"
                >
                  
                  <option
                    value="true"
                  >
                    Aktiviert
                  </option>
                  <option
                    value="false"
                  >
                    Deaktiviert
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          
        </div>
      </div>
      <div
        class="w-full pt-[32px]"
      >
        <div>
          <div
            class="flex items-center w-full bg-white border-border border-[1px] rounded-t-[8px] h-[72px] py-[26px] px-[48px] pl-[23px]"
          >
            <img
              alt="form icon"
              class="object-none pr-[10px]"
              src="[object Object]"
            />
            <h3
              class=""
              id="lblTitle-1"
            >
              Konnektor-Einstellungen
            </h3>
          </div>
          
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.entryOption.hostname"
            >
              <label
                class="text-sm"
                id="form-connector.entryOption.hostname-label"
              >
                Host
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die IP-Adresse oder der Hostname des Konnektors, zu dem Sie eine Verbindung herstellen möchten. Zu finden im Installationsblatt der Praxis (z.B. 127.0.0.1).
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-connector.entryOption.hostname"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.entryOption.port"
            >
              <label
                class="text-sm"
                id="form-connector.entryOption.port-label"
              >
                Port
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die Portnummer, über die der Konnektor erreichbar ist. Standardmäßig ist dies Port 443 für HTTPS.
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-connector.entryOption.port"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.contextParameter.mandantId"
            >
              <label
                class="text-sm"
                id="form-connector.contextParameter.mandantId-label"
              >
                Mandant-ID
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die ID des Infomodels, dass der Konnektor für den Mandanten verwendet. Zu finden in der administrativen Oberfläche des Konnektors oder im Installationsblatt der Praxis.
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-connector.contextParameter.mandantId"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.contextParameter.clientId"
            >
              <label
                class="text-sm"
                id="form-connector.contextParameter.clientId-label"
              >
                Client-ID
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die ID des Infomodels, dass der Konnektor für die Client-ID verwendet.  Zu finden in der administrativen Oberfläche des Konnektors oder im Installationsblatt der Praxis.
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-connector.contextParameter.clientId"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.contextParameter.workplaceId"
            >
              <label
                class="text-sm"
                id="form-connector.contextParameter.workplaceId-label"
              >
                Arbeitsplatz-ID
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die ID des Infomodels, dass der Konnektor für den Arbeitsplatz verwendet.  Zu finden in der administrativen Oberfläche des Konnektors oder im Installationsblatt der Praxis.
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-connector.contextParameter.workplaceId"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.tlsAuthType"
            >
              <label
                class="text-sm"
                id="form-connector.tlsAuthType-label"
              >
                TLS Authentisierung
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Die Methode, die zur Authentisierung verwendet wird, um eine sichere Verbindung zum Konnektor aufzubauen. Dies kann beispielsweise "Zertifikat" oder "Benutzername/Passwort" sein.
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-connector.tlsAuthType"
                >
                  
                  <option
                    value="BasicAuth"
                  >
                    Benutzername/Passwort
                  </option>
                  <option
                    value="ServerClientCertAuth"
                  >
                    Zertifikat PEM-Format
                  </option>
                  <option
                    value="ServerClientCertAuth_Pfx"
                  >
                    Zertifikat P12-Datei (RSA only)
                  </option>
                  <option
                    value="ServerCertAuth"
                  >
                    Keine Authentisierung
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-connector.entryOption.tlsRejectUnauthorized"
            >
              <label
                class="text-sm"
                id="form-connector.entryOption.tlsRejectUnauthorized-label"
              >
                Konnektor Zertifikat prüfen
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Diese Option prüft, ob das Serverzertifikat des Konnektors valide ist und ob es in dem Vertrauensbereich des Truststores liegt.
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-connector.entryOption.tlsRejectUnauthorized"
                >
                  
                  <option
                    value="true"
                  >
                    Aktiviert
                  </option>
                  <option
                    value="false"
                  >
                    Deaktiviert
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          
        </div>
      </div>
      <div
        class="w-full pt-[32px]"
      >
        <div>
          <div
            class="flex items-center w-full bg-white border-border border-[1px] rounded-t-[8px] h-[72px] py-[26px] px-[48px]"
          >
            <!--v-if-->
            <h3
              class=""
              id="lblTitle-2"
            >
              Proxy Einstellungen
            </h3>
          </div>
          
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-proxy.proxySettingsType"
            >
              <label
                class="text-sm"
                id="form-proxy.proxySettingsType-label"
              >
                Proxy-Authentifizierung
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Wenn ein Proxy eingesetzt wird und dieser eine Authentisierung vorsieht, dann können die Einstellung dazu hinterlegt werden. Diese können beispielsweise "Basic Authentifizierung" oder "Zertifikats-Authentifizierung" sein.
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-proxy.proxySettingsType"
                >
                  
                  <option
                    value="false"
                  >
                    Keine Proxy-Authentifizierung
                  </option>
                  <option
                    value="BasicAuth"
                  >
                    Basic Authentifizierung
                  </option>
                  <option
                    value="ProxyClientCert"
                  >
                    Zertifikats-Authentifizierung
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-proxy.useOsSettings"
            >
              <label
                class="text-sm"
                id="form-proxy.useOsSettings-label"
              >
                Betriebssystem Einstellungen benutzen
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Bei Aktivierung dieser Funktion werden automatisch die Proxy-Einstellungen des Betriebssystems verwendet.
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-proxy.useOsSettings"
                >
                  
                  <option
                    value="true"
                  >
                    Aktiviert
                  </option>
                  <option
                    value="false"
                  >
                    Deaktiviert
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-proxy.proxyIgnoreList"
            >
              <label
                class="text-sm"
                id="form-proxy.proxyIgnoreList-label"
              >
                kein Proxy für:
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Per Default laufen alle Adressen über die Windows-Standardeinstellungen der Proxies. Mit dieser Option kann man einstellen, dass bestimmte Adressen nicht über den Standard-Proxy laufen. (Trennzeichen: ";"). Beispiel-Eingaben, die unterstützt werden: 172.*.*.*;*.gematik.de;*.int.gematik.de;127.0.0.1
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-proxy.proxyIgnoreList"
                  placeholder=""
                  type="input"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          
        </div>
      </div>
      <div
        class="w-full pt-[32px]"
      >
        <div>
          <div
            class="flex items-center w-full bg-white border-border border-[1px] rounded-t-[8px] h-[72px] py-[26px] px-[48px]"
          >
            <!--v-if-->
            <h3
              class=""
              id="lblTitle-3"
            >
              Timeout Einstellungen
            </h3>
          </div>
          
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-timeoutValue"
            >
              <label
                class="text-sm"
                id="form-timeoutValue-label"
              >
                Wert des Timeouts in Milli-Sekunden
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Hier kann die Dauer des Timeouts zum Verbindungsaufbau in Millisekunden eingetragen werden (Default 10000ms).
                </span>
              </div>
            </div>
            <div>
              <div>
                <input
                  class="settings-input"
                  id="form-timeoutValue"
                  placeholder="10000"
                  type="number"
                />
              </div>
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          
        </div>
      </div>
      <div
        class="w-full pt-[32px]"
      >
        <div>
          <div
            class="flex items-center w-full bg-white border-border border-[1px] rounded-t-[8px] h-[72px] py-[26px] px-[48px]"
          >
            <!--v-if-->
            <h3
              class=""
              id="lblTitle-4"
            >
              Automatische Updates
            </h3>
          </div>
          
          <div
            class="form-element-container flex justify-between items-center px-[48px] py-[8px] bg-neutral inner-box-shadow"
          >
            <div
              class="text-sm"
              for="input-checkUpdatesAutomatically"
            >
              <label
                class="text-sm"
                id="form-checkUpdatesAutomatically-label"
              >
                Updates automatisch durchführen
              </label>
              <div
                class="tooltip"
              >
                <img
                  alt=""
                />
                <span
                  class="tooltiptext"
                >
                  Ist diese Option "Aktiviert", dann prüft der Authenticator regelmäßig , ob es Updates gibt und führt diese eigenständig durch
                </span>
              </div>
            </div>
            <div>
              <!--v-if-->
              <div>
                <select
                  class="settings-input"
                  id="form-checkUpdatesAutomatically"
                >
                  
                  <option
                    value="true"
                  >
                    Aktiviert
                  </option>
                  <option
                    value="false"
                  >
                    Deaktiviert
                  </option>
                  
                </select>
              </div>
              <!--v-if-->
            </div>
          </div>
          
        </div>
      </div>
      
      <div
        class="flex items-center pt-[32px]"
      >
        <button
          class="bt"
          id="btnSaveSettings"
          title="Die Änderungen sind erst wirksam nach dem Speichern"
          type="submit"
        >
          Speichern
        </button>
        <div
          class="w-3/3"
        >
          <div>
            <button
              class="bt ml-[15px]"
              id="btnFeatureTest"
              type="button"
            >
              Funktionstest
            </button>
            <br />
          </div>
        </div>
        <div
          class="w-3/3"
        >
          <div>
            <button
              class="bt ml-[15px]"
              id="btnLogToZip"
              type="button"
            >
              Log-Daten als ZIP-Datei exportieren
            </button>
            <br />
          </div>
        </div>
      </div>
    </form>
  </div>
  <!--v-if-->
  
</div>
`;
